pywbem/_logging.py

```diff
@@ -14,371 +14,303 @@
 
-          log_filename (:term:`string`)
-            Optional string that defines the filename for output of logs
-            if the dest type is `file`
+    conn = WBEMConnection()
+    conn.configure_api_logger(dest='file', log_filname='xxx.log',
+                              connection=conn)
 
-        Raises:
+Log everything to stderr and output all of the information using the
+pywbem method that configures the loggers from a string::
 
-          ValueError: Generated if the syntax of the input string is invalid
-            or any of the components is not one of allowed strings.
+    define_loggers_from_string('all=stderr':summary', connection=True)
+    conn = WBEMConnection()
 
-        Examples::
+or::
 
-            ops=stderr:min    # set cim operations logger
+    define_loggers_from_string('api=stderr':summary,http=summary',
+                               connection=True)
+    conn = WBEMConnection()
 
-            http=file:        # set http logger to send to file
+Log only http requests and responses including all infomration in the
+requests and responses to a file::
 
-            all=file:all      # Set all loggers to default log destination
-        """  # noqa: E501
-        # pylint: enable=line-too-long
+    define_logger('http', 'file', log_filename='mylog.log')
+    conn = WBEMConnection(..., enable_log='all')
 
-        results = cls._parse_log_specs(input_str)
-        for name, value in six.iteritems(results):
-            cls.create_logger(name, log_dest=value[0],
-                              log_detail_level=value[1],
-                              log_filename=log_filename)
+The logger names themselves are uniquely identified with an id string for
+each WBEMconnection so that the log itself may look like  the followingfor log
+configuration with detail_level = 'summary'::
 
-    @classmethod
-    def create_logger(cls, log_component, log_dest=DEFAULT_LOG_DESTINATION,
-                      log_filename=DEFAULT_LOG_DESTINATION,
-                      log_detail_level=DEFAULT_LOG_DETAIL_LEVEL):
-        """
-        Create the logger defined by the input parameters and place the result
-        in a class level dictionary in this class.
+    2018-03-17 11:39:09,877-pywbem.api.1-32073-Connection:1-32073 WBEMConnection(url='http://localhost', creds=None, default_namespace='root/cimv2')
+    2018-03-17 11:41:13,176-pywbem.api.1-32073-Request:1-32073 EnumerateClasses(ClassName=None, DeepInheritance=None, IncludeClassOrigin=None, IncludeQualifiers=None, LocalOnly=None, namespace=None)
+    2018-03-17 11:41:13,635-pywbem.api.1-32073-Return:1-32073 EnumerateClasses(list of CIMClass; count=103)
 
-        This function can be used to set up all of the named loggers used by
-        pywbem.
+"""   # noqa: E501
+# pylint: enable=line-too-long
 
-        Parameters:
+import logging
+
+# NOTE: cannot use from pywbem import WBEMConnection because it is circular
+# import
+import pywbem
```

Comment:
I agree, but why can we then import the entire pywbem? That should never be done in a submodule of pywbem.

Reply:
maybe try `from pywbem.cim_operations import WBEMConnection`?

Reply:
**Done.** Changed to pywbem.cim_operations.  The key was to avoid from blah import.  That is where the algorithm fails.  The reason for this is to call WBEMConnection.configure_logger from the only function left in this module.